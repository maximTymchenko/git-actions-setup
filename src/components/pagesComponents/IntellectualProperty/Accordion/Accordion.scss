@import '../../../../styles/vars';

.accordion {
    // padding: 0 15px;

    &__wrapper {
        @include section__size;
        margin: 100px auto 80px;
        @include main__padding;
        border-bottom: 2px solid $secondary;

        // @media screen and (max-width: $tablet) {
        //     margin-left: 0;
        // }

        @media screen and (max-width: $mobile) {
            // max-width: 325px;
            // position: relative;
            // left: 10px;
        }
    }

    &__head {
        margin-bottom: 15px;
        display: flex;
        justify-content: space-between;
    }

    &__icon {
        cursor: pointer;

        &--close {
            transform: rotate(45deg);
        }
    }

    &__text-wrapper {
        display: none;

        &--show {
            display: block;
        }
    }

    &__title {
        font-family: 'OktaNeue-Black';
        font-size: 28px;
        font-weight: 900;
        color: $text;
    }

    &__text {
        font-family: 'OktaNeue-Regular';
        font-size: 15px;
        color: $text__body;
        margin-bottom: 20px;
    }
}
