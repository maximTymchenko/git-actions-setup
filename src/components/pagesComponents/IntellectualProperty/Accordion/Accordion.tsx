import React, { FC, useState } from 'react';
import showIcon from '../../../../assets/img/workingTogether/show-icon.svg';
import './Accordion.scss';

const Accordion: FC = () => {
    const toggleActiveIcon = (index: number) => {
        if (positionsArray.positions[index] === positionsArray.activePosition) {
            return 'accordion__icon accordion__icon--close';
        } else {
            return 'accordion__icon';
        }
    };

    const [positionsArray, setPositionsArray] = useState({
        activePosition: null,
        positions: [
            { id: 0, title: 'CCPA' },
            { id: 1, title: 'GDPR' },
            { id: 2, title: 'ISO 27001' },
        ],
    });

    const toggleShowRate = (index: number) => {
        setPositionsArray({ ...positionsArray, activePosition: positionsArray.positions[index] });
    };

    const toggleActiveClassRate = (index: number) => {
        if (positionsArray.positions[index] === positionsArray.activePosition) {
            return 'accordion__text-wrapper accordion__text-wrapper--show';
        } else {
            return 'accordion__text-wrapper';
        }
    };

    return (
        <div data-testid="accordion" className="accordion">
            {positionsArray.positions.map((pos, index) => (
                <div className="accordion__wrapper">
                    <div className="accordion__head">
                        <h3 className="accordion__title">{pos.title}</h3>
                        <img
                            onClick={() => toggleShowRate(index)}
                            className={toggleActiveIcon(index)}
                            src={showIcon}
                            alt="Plus"
                        />
                    </div>
                    <div className={toggleActiveClassRate(index)}>
                        <p className="accordion__text">
                            The California Consumer Privacy Act of 2018 (CCPA) gives Californian
                            residents almost ultimate control over their personal information that
                            you collect about them. Even a single newsletter opt-in form could
                            become a nightmare with lots of devastating lawsuits.
                        </p>
                        <p className="accordion__text">
                            We at Blackbird designed and developed a special solution that embeds to
                            the back end of your product. It is going to be a one-click-job to
                            collect and export collected userâ€™s personal information in a neat PDF
                            file or delete it all upon request.
                        </p>
                    </div>
                </div>
            ))}
        </div>
    );
};

export default Accordion;
