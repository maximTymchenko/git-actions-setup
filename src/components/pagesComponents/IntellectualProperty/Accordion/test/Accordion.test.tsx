import React from 'react';
import Accordion from '../Accordion';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(<Accordion></Accordion>);
    expect(getByTestId('accordion')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<Accordion></Accordion>).toJSON();
    expect(tree).toMatchSnapshot();
});
