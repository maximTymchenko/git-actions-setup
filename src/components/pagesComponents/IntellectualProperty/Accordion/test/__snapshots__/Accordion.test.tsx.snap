// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="accordion"
  data-testid="accordion"
>
  <div
    className="accordion__wrapper"
  >
    <div
      className="accordion__head"
    >
      <h3
        className="accordion__title"
      >
        CCPA
      </h3>
      <img
        alt="Plus"
        className="accordion__icon"
        onClick={[Function]}
        src="test-file-stub"
      />
    </div>
    <div
      className="accordion__text-wrapper"
    >
      <p
        className="accordion__text"
      >
        The California Consumer Privacy Act of 2018 (CCPA) gives Californian residents almost ultimate control over their personal information that you collect about them. Even a single newsletter opt-in form could become a nightmare with lots of devastating lawsuits.
      </p>
      <p
        className="accordion__text"
      >
        We at Blackbird designed and developed a special solution that embeds to the back end of your product. It is going to be a one-click-job to collect and export collected user’s personal information in a neat PDF file or delete it all upon request.
      </p>
    </div>
  </div>
  <div
    className="accordion__wrapper"
  >
    <div
      className="accordion__head"
    >
      <h3
        className="accordion__title"
      >
        GDPR
      </h3>
      <img
        alt="Plus"
        className="accordion__icon"
        onClick={[Function]}
        src="test-file-stub"
      />
    </div>
    <div
      className="accordion__text-wrapper"
    >
      <p
        className="accordion__text"
      >
        The California Consumer Privacy Act of 2018 (CCPA) gives Californian residents almost ultimate control over their personal information that you collect about them. Even a single newsletter opt-in form could become a nightmare with lots of devastating lawsuits.
      </p>
      <p
        className="accordion__text"
      >
        We at Blackbird designed and developed a special solution that embeds to the back end of your product. It is going to be a one-click-job to collect and export collected user’s personal information in a neat PDF file or delete it all upon request.
      </p>
    </div>
  </div>
  <div
    className="accordion__wrapper"
  >
    <div
      className="accordion__head"
    >
      <h3
        className="accordion__title"
      >
        ISO 27001
      </h3>
      <img
        alt="Plus"
        className="accordion__icon"
        onClick={[Function]}
        src="test-file-stub"
      />
    </div>
    <div
      className="accordion__text-wrapper"
    >
      <p
        className="accordion__text"
      >
        The California Consumer Privacy Act of 2018 (CCPA) gives Californian residents almost ultimate control over their personal information that you collect about them. Even a single newsletter opt-in form could become a nightmare with lots of devastating lawsuits.
      </p>
      <p
        className="accordion__text"
      >
        We at Blackbird designed and developed a special solution that embeds to the back end of your product. It is going to be a one-click-job to collect and export collected user’s personal information in a neat PDF file or delete it all upon request.
      </p>
    </div>
  </div>
</div>
`;
